hpobench/benchmarks/surrogates/paramnet_benchmark.py

```diff
@@ -0,0 +1,380 @@
+import logging
+from typing import Union, Dict
+
+import ConfigSpace as CS
+import numpy as np
+
+from hpobench.abstract_benchmark import AbstractBenchmark
+from hpobench.util.data_manager import ParamNetDataManager
+
+__version__ = '0.0.1'
+
+logger = logging.getLogger('Paramnet')
+
+# TODO:
+#  - Return value in case of insufficient budget
+#  - Logging.
+
+
+class _ParamnetBase(AbstractBenchmark):
+
+    def __init__(self, dataset: str,
+                 rng: Union[np.random.RandomState, int, None] = None):
+        """
+        Parameters
+        ----------
+        dataset : str
+            Name for the surrogate data. Must be one of ["adult", "higgs", "letter", "mnist", "optdigits", "poker"]
+        rng : np.random.RandomState, int, None
+        """
+
+        self.dataset = dataset
+        self.n_epochs = 50
+
+        super(_ParamnetBase, self).__init__(rng=rng)
+
+        allowed_datasets = ["adult", "higgs", "letter", "mnist", "optdigits", "poker", "vehicle"]
+        assert dataset in allowed_datasets, f'Requested data set is not supported. Must be one of ' \
+                                            f'{", ".join(allowed_datasets)}, but was {dataset}'
+        logger.info(f'Start Benchmark on dataset {dataset}')
+
+        dm = ParamNetDataManager(dataset=dataset)
+        self.surrogate_objective, self.surrogate_costs = dm.load()
+
+    @staticmethod
+    def convert_config_to_array(configuration: Dict) -> np.ndarray:
+        cfg_array = np.zeros(8)
+        cfg_array[0] = 10 ** configuration['initial_lr_log10']
+        cfg_array[1] = round(2 ** configuration['batch_size_log2'])  # todo: round
+        cfg_array[2] = round(2 ** configuration['average_units_per_layer_log2'])
+        cfg_array[3] = 10 ** configuration['final_lr_fraction_log2']
+        cfg_array[4] = configuration['shape_parameter_1']
+        cfg_array[5] = configuration['num_layers']
+        cfg_array[6] = configuration['dropout_0']
+        cfg_array[7] = configuration['dropout_1']
+
+        return cfg_array.reshape((1, -1))
+
+    @staticmethod
+    def get_configuration_space(seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        """
+
+
+        Parameters
+        ----------
+        seed : int, None
+            Fixing the seed for the ConfigSpace.ConfigurationSpace
+
+        Returns
+        -------
+        ConfigSpace.ConfigurationSpace
+        """
+
+        seed = seed if seed is not None else np.random.randint(1, 100000)
+        cs = CS.ConfigurationSpace(seed=seed)
+        # TODO: in origin benchmark: all uniform float and default is the mid value.
```

Comment:
Then why did you change the default value for this version?

Reply:
good question. I've changed the values to the original default values but wrote them explicitly in the constructor so that it is clear what the default values are.